sdk/ml/azure-ai-ml/tests/dsl/unittests/test_dsl_pipeline.py

```diff
@@ -3951,6 +3952,21 @@ def init_finalize_with_invalid_connection_func(int_param: int, str_param: str):
             set_pipeline_settings(on_init="init_job", on_finalize="finalize_job")
         assert str(e.value) == "Please call `set_pipeline_settings` inside a `pipeline` decorated function."
 
+        # invalid case: set on_init for pipeline component
+        @dsl.pipeline
+        def subgraph_func():
+            node = self.component_func()
+            set_pipeline_settings(on_init=node)  # set on_init for subgraph (pipeline component)
+
+        @dsl.pipeline
+        def subgraph_with_init_func():
+            subgraph_func()
+            self.component_func()
+
+        with pytest.raises(UserErrorException) as e:
```

Comment:
why do we raise error instead of ignoring them? like remove them in _to_rest_object

Reply:
in previously implementation, we do simply ignore them; but new design decides to raise error here.
I understand that we expect users aware of this limitation that on_init/on_finalize is not supported in subgraph.